### DISCUSSION TITLE
Holiday calendar name inconsistencies (Native vs English; Full form vs Short form)

### DISCUSSION THREAD

**Discussion by bact** [FossifyOrg/General-Discussion#419] [General]
Currently the holiday calendar list shows territory names in different conventions:

Some use the name in a language of that territory:
- България _(English: Bulgaria)_
- Éire _(Ireland)_
- 日本 _(Japan)_

Some use English name:
- China _(Native: 中国)_
- Ukraine _(Україна)_

Some use full/official form of the territory name:
- República de Nicaragua _(Short: Nicaragua)_
- República Oriental del Uruguay _(Uruguay)_
- Türkiye Cumhuriyeti _(Turkey)_

Some use normal/short form of the territory name:
- Brasil _(Full: República Federativa de Brasil)_
- Polska _(Rzeczpospolita Polska)_
- Portugal _(Portuguese Republic)_

--

Ideally, the language displaying all territories should be the same as the system language (done through localization).

But as currently we hard coded the territory name strings here: https://github.com/FossifyOrg/Calendar/blob/0330d0e391453e8986928ed35373f0442ae30ffa/app/src/main/kotlin/org/fossify/calendar/activities/MainActivity.kt#L1245 
and can't localize them yet - which is fine for the time being, but at least for now we could agree on how we like to put the territory names.

There are 4 possible options:
1) Native* / Short
2) Native / Long
3) System** / Short
4) System / Long

*Native to the territory (not to the user)
**Currently is English
***I do realize that the selection of language that based on territory can be tricky as in a number of territories, each of them has more than one official language (thus more than one name). For example, Lëtzebuerg / Luxembourg​ / Luxemburg​. But one "neutral" strategy is to pick one with the largest speakers in that territory.

--

As, most of the time, [Unicode Common Locale Data Repository](https://cldr.unicode.org/) uses short form of territory, I think we should go by that standard and consider either (1) or (3) option. They are more common for the general public and make the UI more concise. Thunderbird holiday calendars also use this normal/short form of territory names: https://www.thunderbird.net/en-US/calendar/holidays/

The choice between (1) and (3) depends on our assumption if the user who like to add holidays for a territory should know a language of that territory or not.

For example, to add holidays in Japan, one need to understand that 日本 is Japan.
This may not be the case of foreigners living/traveling in Japan or someone elsewhere who has a business partner in Japan.

A compromise can be Option 5:

5) Native + System / Short

But this option may affect the UI design as the string will be longer.

**Comment by Aga-C** [comment]
Please create a feature request with the solution you prefer the most, with a link to this discussion.

**Comment by bact** [comment]
Thank you. Feature request is at: #119
